synthesizer/process/src/trace/mod.rs

```diff
@@ -193,12 +196,10 @@ impl<N: Network> Trace<N> {
             self.global_state_root.get().ok_or_else(|| anyhow!("Global state root has not been set"))?;
         // Retrieve the fee transition.
         let fee_transition = &self.transitions[0];
-        // Construct the proving tasks.
-        let proving_tasks = self.transition_tasks.values().cloned().collect();
         // Compute the proof.
         let (global_state_root, proof) = Self::prove_batch::<A, R>(
             "credits.aleo/fee (private or public)",
-            proving_tasks,
+            &mut self.transition_tasks,
```

Comment:
Please revert this. We should not be mutating the proving_tasks. What if they call this twice?